app/src/main/java/com/android/unio/ui/components/EventEditComponents.kt

```diff
@@ -50,16 +56,103 @@ import androidx.compose.ui.platform.LocalContext
+ *   locations.
+ * @param initialLocation Location? : Initial location to pre-fill the text field.
+ * @param textFieldTestTag String : Test tag for the text field.
+ * @param dropdownTestTag String : Test tag for the dropdown menu.
+ * @param onLocationSelected (Location) -> Unit : Lambda that is called when a location is selected.
+ */
+@Composable
+fun NominatimLocationPicker(
+    locationSearchViewModel: NominatimLocationSearchViewModel,
+    initialLocation: Location?,
+    textFieldTestTag: String,
+    dropdownTestTag: String,
+    onLocationSelected: (Location) -> Unit
+) {
+  val context = LocalContext.current
+
+  val locationQuery by locationSearchViewModel.query.collectAsState()
+  val locationSuggestions by locationSearchViewModel.locationSuggestions.collectAsState()
+  var showDropdown by remember { mutableStateOf(false) }
+
+  var shouldDisplayInitialLocation by remember { mutableStateOf(true) }
+
+  Box(modifier = Modifier.fillMaxWidth()) {
+    OutlinedTextField(
+        value = if (shouldDisplayInitialLocation) initialLocation?.name ?: "" else locationQuery,
+        onValueChange = {
+          locationSearchViewModel.setQuery(it)
+          shouldDisplayInitialLocation = false
+          showDropdown = true
+        },
+        label = { Text(context.getString(R.string.event_creation_location_label)) },
+        placeholder = { Text(context.getString(R.string.event_creation_location_input_label)) },
+        modifier = Modifier.fillMaxWidth().testTag(textFieldTestTag))
+
+    DropdownMenu(
+        expanded = showDropdown && locationSuggestions.isNotEmpty(),
+        onDismissRequest = { showDropdown = false },
+        properties = PopupProperties(focusable = false),
+        modifier = Modifier.fillMaxWidth().heightIn(max = 200.dp)) {
+          locationSuggestions.take(DROP_DOWN_MAX_ROWS).forEach { location ->
+            DropdownMenuItem(
+                text = {
+                  Text(
+                      text =
+                          location.name.take(DROP_DOWN_MAX_CHARACTERS) +
+                              if (location.name.length > DROP_DOWN_MAX_CHARACTERS)
+                                  context.getString(
+                                      R.string.event_creation_location_dropdown_points)
+                              else "",
+                      maxLines = 1)
+                },
+                onClick = {
+                  locationSearchViewModel.setQuery(location.name)
+                  onLocationSelected(location)
+                  showDropdown = false
+                },
+                modifier = Modifier.padding(8.dp).testTag(dropdownTestTag + location.latitude))
+            HorizontalDivider()
+          }
+
+          if (locationSuggestions.size > DROP_DOWN_MAX_ROWS) {
+            DropdownMenuItem(
+                text = { Text(context.getString(R.string.event_creation_location_dropdown_more)) },
+                onClick = {},
```

Comment:
Empty onClick parameter. Is there really nothing to do on click here (not even a Toast or something) ?

Reply:
Yeah this comes from the implementation that was done in my PR of the `Location` input in the `Event` creation screen. which he simply refactored but did not modify it in anyway (or at least I don't see anything that has been modified).

@Redd87 I suggest you add a `Toast` telling the user to continue writing the address, such that the one that he wants shows up in the first three suggestions. 

Or you can make it such that it adds more suggestions to the `Dropdown`, but I don't think allowing the user to extend the Dropdown infinitely is a good idea. 

We could refine it later on by allowing to add a couple of suggestions, only once, but this is not important for this PR.

The first solution, IMO, is the best, which simply consists in adding a `Toast`. 
Please add it if possible, otherwise LGTM as well.

Reply:
Ok, I'll approve for now then. Thank you for the answer

Reply:
Still it'd be great if @Redd87 could add the `Toast` before merging.

Reply:
@Zafouche ok I'll let you add the toast